mlflow/pipelines/steps/split.py

```diff
@@ -238,6 +244,19 @@ def _run(self, output_directory):
         self.execution_duration = self.run_end_time - run_start_time
         return self._build_profiles_and_card(train_df, validation_df, test_df)
 
+    def _validate_user_code_output(self, pre_split_df, post_split_df):
+        import pandas as pd
+
+        if not isinstance(post_split_df, pd.DataFrame):
+            raise MlflowException(
+                message="The split data is not a DataFrame, please return the correct data"
+            )
+        if len(pre_split_df.columns) != len(post_split_df.columns):
+            raise MlflowException(
+                message="The number of columns post split step are different."
+                "Split filter step is used to filter rows rather than filtering out columns."
+            )
```

Comment:
I don't think this case is possible given the constraint on the user defined filtering fn.

Reply:
Same as below